### PR TITLE
add appLinks for handling schema zapstore to open app details

### LINKED ISSUE #142: zap+store://<app-pub-key> scheme
So that we can link directly to an app and it opens zapstore with that app ready for install

### PR DESCRIPTION
**Pull request by frnandu**
fixes #142

### CODE REVIEW

**Comment by frnandu** [comment]
added a test button on https://yana.do/ linking to zapstore://yana.nostr which this PR should help implement

File: lib/navigation/router.dart [hunk 1]
```diff
@@ -136,6 +135,16 @@ AppLifecycleListener? _lifecycleListener;
 
 final dataLibrariesInitializer = FutureProvider<void>((ref) async {
   await ref.read(initializeFlutterData(adapterProvidersMap).future);
+  appLinks.uriLinkStream.listen((uri) async {
```
Comment (franzaps) [inline lib/navigation/router.dart:138]: This is great, thanks! A few minor things: can you use a linter and formatter? `==` should have space around for example. Also `app` can be final.

Question: this listener is not disposed, is this on purpose?

Comment (frnandu) [reply lib/navigation/router.dart:138]: Formatted and made app final, thanks for reviewing.
I'm not sure where and if this listener needs to be disposed. 
Should  we dispose of it at some point in app lifecycle or should it be always active?
The app_links docs don't mention anything about this.

Comment (franzaps) [reply lib/navigation/router.dart:138]: Thanks for fixing. I'll add a note about disposal for the future (we might need to do it if we allow the app to be relaunched). 

No worries for now but maybe next time: You missed spaces around the `!=` below. Easiest thing is to have Dart format on save (I do on vscode). Also you can do `final app = ...`, the compiler infers the type. A linter will probably catch that.

**Pull request merged**